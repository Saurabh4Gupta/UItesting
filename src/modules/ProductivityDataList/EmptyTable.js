/* eslint-disable react/prop-types */
import React from 'react';
import {
  Paragraph,
  EmptyState,
} from '@dentsu-ui/components';

const EmptyTable = () => {
  return (
    <div
      style={{
        background: '#f8f8f8',
        height: '100vh',
      }}>
      <EmptyState
        layout="horizontal"
        size="small"
        image="create"
        actions={{
          secondary: {
            label: 'Add a new data request',
            url: 'http://www.google.com',
            onClick: () => alert('Secondary action callback fired!'),
            isDisabled: false,
            isExternal: true,
            iconLeft: 'add',
            iconRight: null,
            isLoading: false,
          },
        }}>
        <Paragraph>
          No productivity data requests have been moved to this section yet.
        </Paragraph>
      </EmptyState>
    </div>
  );
};

export default EmptyTable;
