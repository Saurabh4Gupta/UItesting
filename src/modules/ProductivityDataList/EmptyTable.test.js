import React from 'react';
import { shallow } from 'enzyme';
import {  EmptyTable } from './EmptyTable';

describe('<EmptyState />', () => {
  let wrapper;
  const props = {
    defaultText: 'Test',
    handleModal: jest.fn(),
    }
  beforeAll(() => {
    wrapper = shallow(<EmptyTable {...props} />);
  })
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('should call handleDelete', () => {
    wrapper.find('ForwardRef').at(0).prop('actions').secondary.onClick()
    expect(wrapper.props().handleModel).not.toBeNull();
  });
})
