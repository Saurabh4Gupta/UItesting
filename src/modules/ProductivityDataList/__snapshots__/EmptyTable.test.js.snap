// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> should match snapshot 1`] = `
<ForwardRef
  actions={
    Object {
      "secondary": Object {
        "iconLeft": "add",
        "iconRight": null,
        "isDisabled": false,
        "isExternal": true,
        "isLoading": false,
        "label": "Create new data request",
        "onClick": [Function],
      },
    }
  }
  image="create"
  layout="horizontal"
  size="small"
>
  <ForwardRef>
    Test
  </ForwardRef>
</ForwardRef>
`;
